File: client-android/lib/src/test/kotlin/stasis/test/client_android/lib/telemetry/ApplicationInformationSpec.kt
package stasis.test.client_android.lib.telemetry

import io.kotest.core.spec.style.WordSpec
import io.kotest.matchers.shouldBe
import stasis.client_android.lib.telemetry.ApplicationInformation
import java.time.Instant

class ApplicationInformationSpec : WordSpec({
    "A none ApplicationInformation" should {
        "provide no information" {
            ApplicationInformation.none().name shouldBe ("none")
            ApplicationInformation.none().version shouldBe ("none")
            ApplicationInformation.none().buildTime shouldBe (0L)
            ApplicationInformation.none().asString() shouldBe ("none;none;0")
        }
    }

    "A valid ApplicationInformation" should {
        "provide information" {
            val now = Instant.now().toEpochMilli()

            val app = object : ApplicationInformation {
                override val name: String = "test-name"
                override val version: String = "test-version"
                override val buildTime: Long = now
            }

            app.name shouldBe ("test-name")
            app.version shouldBe ("test-version")
            app.buildTime shouldBe (now)
            app.asString() shouldBe ("test-name;test-version;$now")
        }
    }
})

File: layers/src/test/scala/stasis/layers/telemetry/ApplicationInformationSpec.scala
package stasis.layers.telemetry

import java.time.Instant

import stasis.layers.UnitSpec

class ApplicationInformationSpec extends UnitSpec {
  "A none ApplicationInformation" should "provide no information" in {
    ApplicationInformation.none.name should be("none")
    ApplicationInformation.none.version should be("none")
    ApplicationInformation.none.buildTime should be(0L)
    ApplicationInformation.none.asString() should be("none;none;0")
  }

  "A valid ApplicationInformation" should "provide information" in {
    val now = Instant.now().toEpochMilli

    val app = new ApplicationInformation {
      override def name: String = "test-name"
      override def version: String = "test-version"
      override def buildTime: Long = now
    }

    app.name should be("test-name")
    app.version should be("test-version")
    app.buildTime should be(now)
    app.asString() should be(s"test-name;test-version;$now")
  }
}
